src/setuptools_scm/git.py

```diff
@@ -26,9 +27,15 @@ def do_ex(self, cmd):
 
     @classmethod
     def from_potential_worktree(cls, wd):
-        real_wd, _, ret = do_ex("git rev-parse --show-toplevel", wd)
+        wd = os.path.abspath(wd)
+        real_wd, _, ret = do_ex("git rev-parse --show-prefix", wd)
+        real_wd = real_wd[:-1]  # remove the trailing pathsep
         if ret:
             return
+        if not real_wd:
+            real_wd = wd
+        else:
+            real_wd = wd[: len(wd) - len(real_wd) - 1]
```

Comment:
lovely trimdown
please assert the real_wd is actually the tail of the path for pedantic/paranoya

Reply:
Fixed in https://github.com/pypa/setuptools_scm/pull/520/commits/b9cdb54319c2f51548dbebfbbab50fbe09406d84